src/diagnostics/RemoteDebugCommands.ts

```diff
@@ -0,0 +1,174 @@
+            this._progress = p;
+            await this.setRemoteDebug(true, confirmMessage, noopMessage);
+        });
+    }
+
+    public async disableRemoteDebug(): Promise<void> {
+        await this.checkForRemoteDebugSupport();
+
+        const confirmMessage: string = 'The app configuration will be updated to disable remote debugging and restarted. Would you like to continue?';
+        const noopMessage: string = 'The app is not configured for debugging.';
+
+        await vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, async (p: vscode.Progress<{}>): Promise<void> => {
+            this._progress = p;
+            await this.setRemoteDebug(false, confirmMessage, noopMessage);
+        });
+    }
+
+    private reportMessage(msg: string): void {
+        this._outputChannel.appendLine(msg);
+
+        if (this._progress) {
+            this._progress.report({ message: msg });
+        }
+    }
+
+    private async getSiteConfig(): Promise<SiteConfigResource> {
+        if (!this._siteConfig) {
+            this.reportMessage('Fetching site configuration...');
+            this._siteConfig = await this._siteClient.getSiteConfig();
+        }
+
+        return this._siteConfig;
+    }
+
+    private async needUpdateSiteConfig(isRemoteDebuggingToBeEnabled: boolean): Promise<boolean> {
+        const siteConfig: SiteConfigResource = await this.getSiteConfig();
+        return isRemoteDebuggingToBeEnabled !== siteConfig.remoteDebuggingEnabled;
+    }
+
+    private async updateSiteConfig(isRemoteDebuggingToBeEnabled: boolean): Promise<void> {
+        const siteConfig: SiteConfigResource = await this.getSiteConfig();
+
+        siteConfig.remoteDebuggingEnabled = isRemoteDebuggingToBeEnabled;
+        this._siteConfig = siteConfig;
+
+        this.reportMessage('Updating site configuration to set remote debugging...');
+        await this._siteClient.updateConfiguration(siteConfig);
+        this.reportMessage('Updating site configuration done...');
+    }
+
+    private async checkForRemoteDebugSupport(): Promise<void> {
+        const siteConfig: SiteConfigResource = await this.getSiteConfig();
+
+        // So far only node on linux is supported
+        if (!siteConfig.linuxFxVersion.toLowerCase().startsWith('node')) {
+            throw new Error('Azure Remote Debugging is not supported for this instance type');
+        }
+    }
+
+    private async getDebugConfiguration(): Promise<vscode.DebugConfiguration> {
+        const sessionId: string = Date.now().toString();
+        const portNumber: number = await portfinder.getPortPromise();
+
+        // So far only node is supported
+        return {
+            name: sessionId,
+            type: 'node',
+            protocol: 'inspector',
+            request: 'attach',
+            address: 'localhost',
+            port: portNumber,
+            localRoot: vscode.workspace.rootPath,
```

Comment:
`vscode.workspace.rootPath` is deprecated. What's the purpose of 'localRoot'? Can you leave it undefined? Otherwise you should pick something from vsocde.workspaceFolders

Reply:
The purpose is only to map sources from the remote path to the local path. Documentation (e.g. https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_remote-debugging) led me to believe that I needed to set both localRoot and remoteRoot.

Reply:
Yeah in their example they just use "${workspaceFolder}". Can you try that?

rootPath was deprecated because of multi-root workspaces. For example, the user might have their source code in the second folder in the workspace, but rootPath always maps to the first folder

Reply:
This ended up being a bigger problem. "${workspaceFolder}" will not evaluate without the context of a folder to begin with (in launch.json, this would be the folder that the launch.json is in). Because I'm starting the debugger from the app services tree, there is no easy way to know which folder to use.

I put in a check so that it will only try to map sources if there is one folder open. Otherwise it will give up. This works for now, and I will see if there is a better solution.